evm_arithmetization/src/cpu/kernel/tests/transaction_parsing/parse_type_0_txn.rs

```diff
@@ -86,6 +86,10 @@ fn process_type_0_txn_invalid_sig() -> Result<()> {
         None,
     );
 
+    // If we reach process_normalized_txn, the test fails (we should have had a
+    // kernel panic beforehand).
+    interpreter.halt_offsets.push(process_normalized_txn);
```

Comment:
I don't understand this change

Reply:
I think it would just allow the test to fail a bit earlier.

Reply:
It wouldn't, the run will abort in `decode_and_store_s` because of the failed assertion, returning a `KernelPanic` early which is then caught below with the `assert!(result.is_err());`. It is already present for the normal run because in that case indeed we want to stop early and not have to go through the rest of the execution (as it would fail, because we didn't set up the state)

Reply:
Ah, yes, it's a remnant of when I misunderstood what `parse_type_X_txn_invalid_sig` was testing. At first I thought it checked that either `ecrecover` panicked, or that the recovered public key didn't match the sender.

Reply:
I stand corrected.